connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -383,6 +383,21 @@ class Dataset[T] private[sql] (
     }
   }
 
+  // TODO(SPARK-50134): Support Lateral Join API in Spark Connect
```

Comment:
According to the recent discussion on `Spark Connect`, is this considered a blocker issue for Apache Spark 4.0.0, @ueshin , @hvanhovell , and @HyukjinKwon ?

I believe this newly added API incompatibility (SPARK-50134) is a nice-to-have, and should be a blocker issue for Apache Spark 4.0.0.

Reply:
IMO I think so ... but I think we should be fine in general ... as usually API are implemented in both Spark Classic and Connect at the same PR or at the similar timeline.